src/server/routes/collection.js

```diff
@@ -80,9 +140,23 @@ router.get('/:collectionId/edit', auth.isAuthenticated, auth.isCollectionOwner,
 
 router.post('/:collectionId/edit/handler', auth.isAuthenticatedForHandler, auth.isCollectionOwner, collectionCreateOrEditHandler);
 
-router.get('/:collectionId', (req, res) => {
+router.post('/:collectionId/remove', auth.isAuthenticated, auth.isCollectionOwnerOrCollaborator, async (req, res) => {
+	const {bbids} = req.body;
 	const {collection} = res.locals;
-	res.status(200).send(collection);
+	try {
+		const {UserCollectionItem} = req.app.locals.orm;
+		const promiseArray = bbids.map((bbid) =>
+			new UserCollectionItem()
+				.where('bbid', bbid)
+				.where('collection_id', collection.id)
+				.destroy()
+		);
+		await Promise.all(promiseArray);
```

Comment:
I wonder if you could do this in one batch instead of multiple calls?
Perhaps using Bookshelf's collections ?